app/jobs/solid_cable/trim_job.rb

```diff
@@ -2,8 +2,20 @@
 
 module SolidCable
   class TrimJob < ActiveJob::Base
-    def perform
-      ::SolidCable::Message.trimmable.delete_all
+    def perform(id = nil)
+      id ||= ::SolidCable::Message.maximum(:id)
+
+      return unless (id % (trim_batch_size / 2)).zero?
```

Comment:
> I think it would be better to do this check before creating the job, as we'll end up with a lot of no-op jobs otherwise.

The job is running as perform_now so we shouldn't be enqueuing anything into the queue and we dont pay a penalty for no-oping. 

> We might be better using a random value rather than id, I could imagine cases say where there are not a lot of message coming through and you have a bunch of processes getting the same value for id

Ahh ok ill check that out

Reply:
> The job is running as perform_now so we shouldn't be enqueuing anything into the queue and we dont pay a penalty for no-oping.

Oh sorry, I missed that 🙈

Reply:
No worries!